\documentclass[12pt,oneside,onecolumn,a4paper]{article}
\setlength{\parindent}{1cm}
\usepackage{indentfirst}
\usepackage[breaklinks=true]{hyperref}
\usepackage{tikz}
\usepackage{pgfplots}
\usepackage[english]{babel}
\usepackage{natbib}
\setlength{\bibhang}{2em}
\usepackage[affil-it]{authblk} 
\usepackage{etoolbox}
\usepackage{lmodern}
\usepackage{caption,setspace}
\usepackage{svg}
\usepackage{float}
\captionsetup{font={small,it,stretch=1.25},labelfont=bf}

\renewcommand\Authfont{\fontsize{14}{14}\selectfont}
\renewcommand\Affilfont{\fontsize{12}{12}\itshape}

\renewcommand{\baselinestretch}{2.0}
\setlength{\parindent}{3em}
\usetikzlibrary{matrix,chains,positioning,decorations.pathreplacing,arrows,datavisualization,datavisualization.formats.functions}
\DeclareGraphicsExtensions{.pdf,.PDF,.png,.PNG,.jpg,.JPG,.jpeg,.JPEG}
\usepackage[a4paper,top=1in,bottom=1in,left=1in,right=1in,marginparwidth=0cm]{geometry}

\title{Unsupervised Analysis of Gene Expression in Neurological Animal Models\vspace{-0.4cm}}
\author{Kevin Hu\vspace{-0.4cm}}
\affil{Massachusetts Academy of Math and Science\vspace{-0.4cm}}
\date{}

\begin{document}


\maketitle

\begin{abstract}
   Coming soon to a STEM fair near you.
\end{abstract}

\break
\tableofcontents
\break

\section{Introduction}

The use of animal models is an essential component of the drug development process. Animal models, in particular mouse models, allow diseases to be studied and the effects of candidate drugs to be observed in real time. This is crucial in neurological research, where mice serve as filters to clinical trials. 

However, genetic differences between humans and animal models confound drug tests. Drug-gene interactions result when the results of a gene interfere with the action of the drug, such as in blockages of drug metabolism, mutated receptor proteins, and general physiological differences. Furthermore, different patterns of spatial and temporal expression between humans and the model organism are also factors in development.

Clustering, or the unsupervised categorization of data points into discrete sets, is widely used in gene expression analysis. However. existing clustering algorithms rely heavily on measures of correlation that are prone to outlier leverage and high sample variance, both of which are common in biological samples. Recently, neural network algorithms have shown promise in image clustering, such as that of the popular MNIST and USPS digit databases. The goal of this project was to engineer a reccurent convolutional neural network for the clustering of region-time expression matrices between human and mouse genes present in cerebral development. 

\section{Literature review}

\subsection{Neurological disease modeling}\citep{Burns_2015}\citep{Lin_2014}


\begin{figure}[h!]
\begin{center}
\includegraphics[width=0.8\columnwidth]{figures/cerebral_organoid/cerebral-organoid-for-Broad-web-300x218}
\caption{A cerebral organoid.\citep{nguyen_wang_nikolakopoulou_2015}%
}
\end{center}
\end{figure}

\subsection{The developing brain}\citep{Bakken_2015}


\begin{figure}[h!]
\begin{center}
\includegraphics[width=0.8\columnwidth]{figures/160913_thompson_atlas/160913_thompson_atlas}
\caption{Stages of mouse brain development. \citep{Thompson_2014}%
}
\end{center}
\end{figure}

\subsection{Gene expression regulation}
Gene expression at the cellular level is regulated by a plethora of molecular means and is the foundation of an organism's development, growth, and survival. Although the vast majority of cells (except germ and immune cells) in an animal have identical DNA, their own structure and function is determined by these methods of regulation. This cellular regulation further determines the structure and function of the upper-level tissues, organs, and ultimately the organism itself. 

Gene expression is accomplished through the processes of DNA transcription to RNA and RNA translation to the amino acid chain that becomes the protein. Transcription is influenced by transcription factors, which affect how well an RNA polymerase (the enzyme primarily responsible for transcription) is able to bind to and form a transcription complex on a gene. Specific examples of molecules affecting gene expression include the addition of acetyl or methyl groups to the DNA and proteins that by binding to the DNA, can either attract or block the binding of RNA polymerase. In eukaryotic cells such as those of mice and humans, the RNA transcript must be further processed prior to transcription. This post-transcriptional processing involves the addition of cap and tail sequences that protect the sequence from digestive enzymes and allow for translation to occur. The RNA transcript also contains exon and intron regions, with only the exon regions being incorporated into the final sequence to be translated. Once the RNA transcript has left the nucleus of the cell (through a nuclear pore), a ribosome (a large cellular machine that facilitates translation of RNA to protein) binds to the RNA and initiates construction of the amino acid chain. Once the amino acid chain is complete, the ribosome splits in two, and the protein may be further modified. For instance, the chain may be folded within a chaperonin, which provides an optimal environment for some proteins, or it may be tagged with small molecules that identify its destination. These methods, and many others, allow for extremely precise regulation of complex genetic pathways that are the drivers of embryonic development.

The embryonic phase of development is known for highly dynamic expression of genes across time and space, or spatiotemporal regulation. Differential gene expression over space and time is the method that allows an organism to develop specialized cells. Conditional regulation of only a  select few high-level "master genes" can determine the fate of a cell. 
\citep{bisceglia_2010}

\begin{figure}[h!]
\begin{center}
\includegraphics[width=0.8\columnwidth]{figures/Untitled/Untitled}
\caption{\textbf{A.} A general overview of the flow of information from DNA to RNA to protein. \textbf{B.} Demonstration of cell differentiation. \citep{bisceglia_2010}.%
}
\end{center}
\end{figure}

\subsection{Gene expression measurement}
Numerous methods exist for the quantification and visualization of gene expression data. In this study, the gene expression levels are computed from either \textit{in situ} hybridization (ISH) or RNA sequencing (RNA-Seq). 

ISH is a technique that allows specific sequences of nucleic acids to be located in a section of cells or tissue. This localization is accomplished through the principle that complementary nucleic acid sequences form hydrogen bonds (hybridize) with each other. The complementary sequence that detects the target sequence, or the \textit{probe}, may be identified through fluorescent (fluorescent DNA components), radioactive (radioactive isotopes), or immunohistochemical (labeled with an antibody) means. After the probe is allowed to hybridize to the target strand, RNases (RNA-digesting enzymes) hydrolyze any excess probes and the rest are washed off, leaving only bound probes. Following ISH, fluorescent microscopy may then be used to highlight regions of gene expression, which may then be scanned and analyzed using image analysis algorithms. \citep{Angerer_1991}

In addition to ISH, RNA-Seq is also used in order to accurately measure gene expression. Because the primary function of RNA in the cell is as an intermediate in the transfer of information from DNA to protein, RNA expression levels can give an approximate indication of overall gene expression levels. In RNA-Seq, as the name suggests, next-generation nucleic acid sequencing is applied to a collection of RNA sampled from a tissue region. The RNA sample is first isolated from the source tissue through digestion using detergents and enzymes. Once the RNAs are isolated, a next-generation sequencing machine reads all the RNA to a digital storage device. These raw sequence data are then formatted to identify exons and introns, which may be later used to identify gene expression levels. 
\citep{Wang_2009}


\begin{figure}[h!]
\begin{center}
\includegraphics[width=0.8\columnwidth]{figures/ISH/ISH}
\caption{An ISH stain for glial fibrillary acidic protein (GFAP) RNA in the P14 mouse brain.%
}
\end{center}
\end{figure}

\subsection{Gene expression analysis}


\subsection{The Allen Brain Atlas}
Founded in 2003 by Microsoft co-founder Paul Allen, the Allen Institute for Brain Science is a nonprofit research organization dedicated to the public pursuit of brain research. Among the many free datasets provided by the Institute are the Allen Developing Mouse Brain Atlas (located at \href{http://www.developingmouse.brain-map.org/}{www.developingmouse.brain-map.org/}) and the Allen Atlas of the Developing Human Brain (located at \href{http://www.brainspan.org/}{www.brainspan.org/}).

The Developing Mouse Brain Atlas features in situ hybridization (ISH) data for over 2,100 genes across multiple stages of embryonic and postnatal mouse brain development. The raw images of the ISH scans total 434,946 images. These images were assembled into 3-dimensional grids of expression values.  In addition, an application programming interface (API) in the form of a bioinformatics pipeline allows researchers to access all data produced by the Developing Mouse Brain studies. The Developing Mouse API further allows researchers to obtain quantized expression values per region, which are calculated based off of analysis of the ISH scan images. These expression values may be found for several different developmental timepoints, at several discrete brain regions, for each of the 2,100+ genes. \citep{Thompson_2014}

Similar to the Developing Mouse Brain Atlas, the Developing Human Brain Atlas also features a diverse array of ISH expression values for several thousand genes. These expression values are available for download as raw comma-separated-values (CSV) files from the Developing Human Brain Atlas Website, although an API is still available. In addition to ISH images, extremely detailed, cellular-level, magnetic resonance imaging (MRI) and microarray data are also included. \citep{24695229}

\begin{figure}[h!]
\begin{center}
\includegraphics[width=0.8\columnwidth]{figures/Abl1/Abl1}
\caption{A heatmap of expression values for the Abelson murine leukemia viral oncogene homolog 1 (ABL1) gene in the developing mouse brain. Expression values were obtained from ISH stains and computed for each region using specialized 3-D voxel counts provided by the Allen Brain Atlas API. The y-axis is the development stage and the x-axis is the brain region (abbreviated). The expression values were transformed to a logarithmic scale to account for skew.%
}
\end{center}
\end{figure}

\subsection{Neural networks}
In 1959, Arthur Samuel described machine learning as the ``field of study that gives computers the ability to learn without being
explicitly programmed," a definition that continues to hold true. Today neural networks are one of the most flexible and powerful machine learning algorithms. In several areas of research, neural networks offer state-of-the-art performance over traditional, hand-coded methods, such as in image pattern recognition, genomic analysis, and protein folding. The basic neural network node, or \textit{neuron}, is similar to the biological neural network in that it receives a number of inputs and 
\begin{figure}[H]
\begin{center}
\includegraphics[width=0.8\columnwidth]{figures/activations}
\caption{Graphs of common neuron activation functions.\label{fig:activations}}
\end{center}
\end{figure}

\begin{table}[H]
\begin{center}
\caption{Equations of the activation functions shown in Figure~\ref{fig:activations}.}
\begin{tabular}{ l | l | l | l }
 Linear & Sigmoid & Tanh & ReLU\\ 
 $a(x) = x$ & $a(x) = \frac{1}{1+e^{-x}}$ & $a(x) = \frac{e^{x}-e^{-x}}{e^{x}+e^{-x}}$ & $a(x) = 0\ \textrm{if}\ x \leq 0, x\ \textrm{if}\ x > 0$\\   
\end{tabular}
\label{table:1}
\end{center}
\end{table}
Neurons output a numerical value based on an activation function.

\begin{figure}[H]
\begin{center}
\includegraphics[width=0.8\columnwidth]{figures/doodle}
\caption{An example neural network that learns to generate an image given x and y pixel position inputs. The inputs are the x and y coordinates of the pixel, and the outputs are the three red-green-blue (RGB) values that define the color of a pixel. A few results of this example network are shown in Figure~\ref{fig:doodle_results}.
\label{fig:doodle}
}
\end{center}
\end{figure}

\begin{figure}[H]
\begin{center}
\includegraphics[width=0.8\columnwidth]{figures/doodle_images}
\caption{The results of an neural network outlined in Figure~\ref{fig:doodle} trained to reproduce images by outputting an RGB triple based on x-y coordinates.
\label{fig:doodle_results}
}
\end{center}
\end{figure}
\begin{figure}[H]
\begin{center}
\includegraphics[width=0.8\columnwidth]{figures/convnet}
\caption{An example of a simple convolutional network designed for digit classification of the MNIST dataset. }%
\end{center}
\end{figure}

\begin{figure}[H]
\begin{center}
\includegraphics[width=0.8\columnwidth]{figures/CNN/CNN}
\caption{An example of a multi-layer CNN for classifying handwritten digits from the MNIST dataset. The feature extraction section repeatedly convolves the input image to higher level features, which are fed as inputs to the classification stage. In this case, the classification stage is composed of two fully connected layers, with the final layer outputting the predicted digit classification. \citep{peemen_mesman_corporaal_2011}%
}
\end{center}
\end{figure}

\subsection{Unsupervised learning}


\subsection{Clustering}

\begin{figure}[H]
\begin{center}
\includegraphics[width=0.8\columnwidth]{figures/CNN/CNN}
\caption{An example of a multi-layer CNN for classifying handwritten digits from the MNIST dataset. The feature extraction section repeatedly convolves the input image to higher level features, which are fed as inputs to the classification stage. In this case, the classification stage is composed of two fully connected layers, with the final layer outputting the predicted digit classification. \citep{peemen_mesman_corporaal_2011}%
}
\end{center}

\citep{yangCVPR2016joint}

\section{Research plan}


\subsection{Researchable question}
Are there significant differences between patterns of gene expression in developing mouse and human central nervous systems?

\subsection{Hypothesis}

It is hypothesized that there exist significant differences between the gene expression profiles of developing mouse and human central nervous systems.

\subsection{Procedure}
This project will be entirely performed on a computer. Gene expression profiles for developing mouse and human brains will be first downloaded from the Allen Brain Atlas. Next, the data will be formatted and transformed into matrices of region versus time expression expression matrices. At this point, there will be two datasets, a developing human dataset and a developing mouse dataset. 


For each dataset, a convolutional neural network algorithm will be applied in order to automatically classify genes into discrete clusters based on patterns of expression. The relative positions of genes in the human and mouse clusterings will then be examined. A different clustering would suggest that the respective transcriptomic landscape is different. The gene expression clusterings may be applied to highlight discrete drug-gene interactions in human versus mouse brains. Potential differing expressions may account for anatomical differences in mouse and human brains. Time permitting, the similarity of expression patterns in cerebral organoids will also be considered. The insights from this project may also be applied to single-cell cancer sequencing, where clusterings may be used to cluster and identify stages of cancer cell development.

\section{Methodology}

\section{Results}

\section{Discussion}

\section{Conclusions}

\section{Limitations}

\section{Extensions}

\section{Acknowledgements}

\bibliographystyle{apa}
\bibliography{bibliography/biblio.bib%
}

\end{document}

